crates/autopilot/src/database/competition.rs

```diff
@@ -67,7 +68,11 @@ pub struct Competition {
 }
 
 impl super::Postgres {
-    pub async fn save_competition(&self, competition: &Competition) -> anyhow::Result<()> {
+    pub async fn save_competition(
```

Comment:
Given the two things are orthogonal, can we make this a separate method, `save_fee_policies` (and maybe start adding the fee policy persistence into the `infra` module)

Reply:
Ok, will do. But this still should be a part of the same DB transaction, right? I was worried about the inconsistency.

Reply:
The data that is being written is completely different though, isn't it? Other than performance, is there any advantage of sharing the transaction?

Reply:
Migrated to infra

Reply:
We should probably first merge https://github.com/cowprotocol/services/pull/2188 since a lot of refactoring is done there so no point moving any of the functionality from this PR to `infra` without first looking into ☝️ PR.